File: app/common/CryptoUtils.java
package common;

import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class CryptoUtils {
  public static String md5(String input) throws NoSuchAlgorithmException {
    String result = null;
    MessageDigest md = MessageDigest.getInstance("MD5");
    md.update(input.getBytes());
    BigInteger hash = new BigInteger(1, md.digest());
    result = hash.toString(16);
    while(result.length() < 32) {
      result = "0" + result;
    }
    return result;
  }
}

File: app/common/CryptoUtils.scala
package common
import org.apache.commons.codec.digest.DigestUtils
import java.io.InputStream

/**
 * User: ub (13/7/13 - 12:48 PM)
 */
object CryptoUtils {
  def md5(input: String): String = DigestUtils.md5Hex(input)
  def md5(input: InputStream): String = DigestUtils.md5Hex(input)
}
